src/ai/backend/manager/registry.py

```diff
@@ -204,6 +206,58 @@
 SESSION_NAME_LEN_LIMIT = 10
 
 
+async def get_known_container_registries(
+    db: ExtendedAsyncSAEngine,
+) -> Mapping[str, Mapping[str, yarl.URL]]:
+    async with db.begin_readonly_session() as db_session:
+        registries: list[tuple[str, str, str]] = (
+            await db_session.execute(
+                sa.select([
+                    ContainerRegistryRow.project,
+                    ContainerRegistryRow.registry_name,
+                    ContainerRegistryRow.url,
+                ])
+            )
+        ).fetchall()
+
+        result: MutableMapping[str, MutableMapping[str, yarl.URL]] = {}
+
+        for registry_row in registries:
+            project, registry_name, url = registry_row
+
+            if project not in result:
+                result[registry_row[0]] = {}
+
+            result[project][registry_name] = yarl.URL(url)
+
+        return result
```

Comment:
```suggestion
    async with db.begin_readonly_session() as db_session:
        
        query_stmt = (
          sa.select(ContainerRegistryRow)
          .options(load_only(
                    ContainerRegistryRow.project,
                    ContainerRegistryRow.registry_name,
                    ContainerRegistryRow.url
                    ))
        )
        registries = cast(list[ContainerRegistryRow], (await db_session.scalars(query_stmt).all())
        result: MutableMapping[str, MutableMapping[str, yarl.URL]] = {}

        for registry_row in registries:
            project = registry_row.project
            registry_name = registry_row.registry_name
            url = registry_row.url

            if project not in result:
                result[project] = {}

            result[project][registry_name] = yarl.URL(url)

        return result
```

Reply:
Just a comment. how about adding this function to `ContainerRegistryRow` as a classmethod?